packages/backend/api/src/cocktail.ts

```diff
@@ -48,190 +55,218 @@ cocktailRouter.get('/', async (req, res) => {
-      const ingredientList: string[] = Array.isArray(ingredients)
-        ? ingredients.map(String)
-        : [String(ingredients)];
-
-      const countDistinctIngredients = sql<string>`COUNT(DISTINCT ingredient.name)`;
-
-      query = query
-        .having(countDistinctIngredients, '=', sql`${ingredientList.length}`)
-        .where('ingredient.name', 'in', ingredientList);
-    }
-
-    if (flavours !== undefined) {
-      const flavourList: Flavour[] = (
-        Array.isArray(flavours) ? flavours.map(String) : [String(flavours)]
-      ) as Flavour[];
-
-      const countDistinctFlavours = sql<string>`COUNT(DISTINCT cocktail.final_flavour)`;
-
-      query = query
-        .having(countDistinctFlavours, '=', sql`${flavourList.length}`)
-        .where('cocktail.final_flavour', 'in', flavourList);
-    }
-
-    if (kcals !== undefined) {
-      const kcalList: number[] = Array.isArray(kcals)
-        ? kcals.map(Number)
-        : [Number(kcals)];
-
-      const countDistinctKcals = sql<string>`COUNT(DISTINCT cocktail.total_kcal)`;
-
-      query = query
-        .having(countDistinctKcals, '=', sql`${kcalList.length}`)
-        .where('cocktail.total_kcal', 'in', kcalList);
-    }
-
-    if (complexities !== undefined) {
-      const complexityList: number[] = Array.isArray(complexities)
-        ? complexities.map(Number)
-        : [Number(complexities)];
-
-      const countDistinctComplexity = sql<string>`COUNT(DISTINCT recipe.total_complexity)`;
-
-      query = query
-        .having(countDistinctComplexity, '=', sql`${complexityList.length}`)
-        .where('recipe.total_complexity', 'in', complexityList);
+cocktailRouter.get(
+  '/alcohol',
+  checkAuthState,
+  async (req: RequestWithUser, res) => {
+    const ingredients = req.query.ingredients;
+    const flavours = req.query.flavours;
+    const kcals = req.query.kcals;
+    const complexities = req.query.complexities;
+    const degrees = req.query.degrees;
+    const searchTerm = req.query.searchTerm;
+
+    const userId = req.userId;
+    const isLogin = req.login;
+
+    try {
+      let query = db
+        .selectFrom('cocktail')
+        .innerJoin('recipe', 'recipe.cocktail_id', 'cocktail.id')
+        .innerJoin('action', 'recipe.action_id', 'action.id')
+        .innerJoin(
+          'action_ingredient',
+          'action.id',
+          'action_ingredient.action_id',
+        )
+        .innerJoin(
+          'ingredient',
+          'action_ingredient.ingredient_id',
+          'ingredient.id',
+        )
+        .leftJoin('favorite', 'favorite.cocktail_id', 'cocktail.id')
+        .select([
+          'cocktail.id as cocktail_id',
+          'cocktail.name as cocktail_name',
+          'cocktail.image as cocktail_image',
+          'cocktail.ratings_average as avg_rating',
+          'cocktail.created_at as cocktail_created',
+          'cocktail.final_flavour as cocktail_flavour',
+          'cocktail.total_kcal as cocktail_kcal',
```

Comment:
une raison de renommer chaque colonne ?

Reply:
c grave docteur ?

Reply:
je vais pas y toucher, c'est le code de Justine (que tu as accepter by the way). si j'y touche autant toucher a ça page car je vais tout casser..